carma_wm/src/collision_detection.cpp

```diff
@@ -0,0 +1,179 @@
+            
+            collision_detection::MovingObject v;
+
+            geometry_msgs::Pose pose;
+            pose.position.x = tp.trajectory_points[0].x;
+            pose.position.y = tp.trajectory_points[0].y;
+
+            Eigen::Vector2d vehicle_vector = {tp.trajectory_points[0].x - tp.trajectory_points[1].x , tp.trajectory_points[0].y - tp.trajectory_points[1].y};
+
+            Eigen::Vector2d x_axis = {1, 0};
+
+            double yaw = std::acos(vehicle_vector.dot(x_axis)/(vehicle_vector.norm() * x_axis.norm()));
+
+            tf2::Quaternion vehicle_orientation;
+            vehicle_orientation.setRPY(0, 0, yaw);
+
+            pose.orientation.x = vehicle_orientation.getX();
+            pose.orientation.y = vehicle_orientation.getY();
+            pose.orientation.z = vehicle_orientation.getZ();
+            pose.orientation.w = vehicle_orientation.getW();
+
+            v.object_polygon = CollisionChecking::ObjectToBoostPolygon<polygon_t>(pose, size);
+            v.linear_velocity = veloctiy.linear;
+
+            for(int i=0; i < tp.trajectory_points.size() - 1; i++){
+
+                vehicle_vector = {tp.trajectory_points[i + 1].x - tp.trajectory_points[i].x , tp.trajectory_points[i+1].y - tp.trajectory_points[i].y};
+                yaw = std::acos(vehicle_vector.dot(x_axis)/(vehicle_vector.norm() * x_axis.norm()));
+
+                tf2::Quaternion orientation;
+                orientation.setRPY(0, 0, yaw);
+
+                geometry_msgs::Pose pose;
+                pose.position.x = tp.trajectory_points[i].x;
+                pose.position.y = tp.trajectory_points[i].y;
+
+                pose.orientation.x = orientation.getX();
+                pose.orientation.y = orientation.getY();
+                pose.orientation.z = orientation.getZ();
+                pose.orientation.w = orientation.getW();
+
+                v.future_polygons.push_back(CollisionChecking::ObjectToBoostPolygon<polygon_t>(pose, size));
+            }
+
+            return v;
+        };
+
+        bool CollisionChecking::DetectCollision(collision_detection::MovingObject ob_1, collision_detection::MovingObject ob_2, int target_time) {            
+
+            collision_detection::MovingObject ob_1_after = CollisionChecking::PredictObjectPosition(ob_1,target_time);
+            collision_detection::MovingObject ob_2_after = CollisionChecking::PredictObjectPosition(ob_2,target_time);
+
+            if(CollisionChecking::CheckPolygonIntersection(ob_1_after, ob_2_after)){
+                return true;
+            }
+            
+            return false;
+        };
+
+        bool CollisionChecking::CheckPolygonIntersection(collision_detection::MovingObject ob_1, collision_detection::MovingObject ob_2) {    
+
+                std::deque<polygon_t> output;
+
+                boost::geometry::intersection(ob_1.object_polygon, ob_2.object_polygon, output); 
+
+                BOOST_FOREACH(polygon_t const& p, output)
```

Comment:
seems like you could just use a regular C++ 11 for each loop here instead of the boost macro

```
for (auto p : output) {
// logic
}
```